test/models/maintenance_tasks/run_test.rb

```diff
@@ -690,6 +702,17 @@ class RunTest < ActiveSupport::TestCase
 
     private
 
+    def assert_notification_for(name, expected_payload)
```

Comment:
Small improvement: You should be able to use `ActiveSupport::Notifications.subscribed` which takes a block and unsubscribes after.

https://api.rubyonrails.org/classes/ActiveSupport/Notifications.html